src/bp-core/bp-core-functions.php

```diff
@@ -2791,7 +2791,22 @@ function bp_nav_menu_get_loggedin_pages() {
 	$bp_menu_items = array();
 
 	if ( 'rewrites' !== bp_core_get_query_parser() ) {
-		$bp_menu_items = $bp->members->nav->get_primary();
+		$primary_items = $bp->members->nav->get_primary();
+
+		foreach( $primary_items as $primary_item ) {
+			$current_user_link = $primary_item['link'];
+
+			// When displaying a user, reset the primary item link.
+			if ( bp_is_user() ) {
```

Comment:
Could this be moved outside of the `foreach`?

Reply:
This is actually the main part of the fix. Do you mean using a variable before the `foreach` ?

Eg:
```php
$is_user_displayed = bp_is_user();

foreach () {
      if ( $is_user_displayed ) {}
}
```

Reply:
Correct!

Reply:
I'm not sure how it'd break something... 🤔

Reply:
Ok I'll update the PR 👍